configure-android.sh

```diff
@@ -1,199 +1,218 @@
-    --)
-        shift ; break ;;
-
-    # illegal switch
-    *)
-        echo -e \\n"Option -$OPTARG not allowed."
-        HELP
-        exit 1
+        --libpcap-include-dir)
+            LIBPCAP_INCLUDE_DIR=$2
+            shift 2
+            ;;
+        --libpcap-lib-dir)
+            LIBPCAP_LIBRARY_DIR=$2
+            shift 2
+            ;;
+        *)  exit_with_error "Unrecognized option: $1"
     esac
 done
 
-if [ -z "$NDK_PATH" ]; then
-    echo "Please specify the NDK path using with '--ndk-path'. Exiting..."
-    exit 1
+if [ -z "$ANDROID_NDK_ROOT" ] ; then
+    exit_with_error "Please specify the Android NDK root path via --ndk-path PATH"
 fi
 
-if [ -z "$TARGET" ]; then
-    echo "Please specify the target using wity '--target'. Exiting..."
-    exit 1
+if [ -z "$ANDROID_API_LEVEL" ] ; then
+    ANDROID_API_LEVEL=29
+fi
+
+if [ -z "$TARGET_TRIPLE" ] ; then
+    exit_with_error "Please specify the target via --target TARGET"
 fi
 
 if [ -z "$LIBPCAP_INCLUDE_DIR" ]; then
-    echo "Please specify the location of libpcap header files with '--libpcap-include-dir'. Exiting..."
-    exit 1
+    exit_with_error "Please specify the directory of libpcap header files via --libpcap-include-dir DIR"
 fi
 
-if [ -z "$LIBPCAP_LIB_DIR" ]; then
-    echo "Please specify the location of libpcap binary that matches the reqruied arch with '--libpcap-lib-dir'. Exiting..."
-    exit 1
+if [ -z "$LIBPCAP_LIBRARY_DIR" ]; then
+    exit_with_error "Please specify the directory of libpcap binary that matches the specified target via --libpcap-lib-dir DIR"
 fi
 
-PLATFORM_MK="mk/platform.mk"
-PCAPPLUSPLUS_MK="mk/PcapPlusPlus.mk"
+#########################################################################################
+
+unset BUILD_MACHINE_OS_TYPE
+unset BUILD_MACHINE_OS_ARCH
+
+BUILD_MACHINE_OS_TYPE=$(uname | tr A-Z a-z)
+BUILD_MACHINE_OS_ARCH=$(uname -m)
+
+ANDROID_TOOLCHAIN=$ANDROID_NDK_ROOT/toolchains/llvm/prebuilt/$BUILD_MACHINE_OS_TYPE-$BUILD_MACHINE_OS_ARCH
+
+#########################################################################################
+
+cat <<EOF
+Build machine OS type: $BUILD_MACHINE_OS_TYPE
+Build machine OS arch: $BUILD_MACHINE_OS_ARCH
+
+Android NDK root:  $ANDROID_NDK_ROOT
+Anrdoid API level: $ANDROID_API_LEVEL
+Android toolchain path: $ANDROID_TOOLCHAIN
+
+Target triple: $TARGET_TRIPLE
+
+libpcap include dir: $LIBPCAP_INCLUDE_DIR
+libpcap library dir: $LIBPCAP_LIBRARY_DIR
+EOF
+
+#########################################################################################
+
+unset TEMP_DIR
+unset TEMP_CONFIG_FILE
 
-# copy the basic Android platform.mk
-cp -f mk/platform.mk.android $PLATFORM_MK
+TEMP_DIR=$(mktemp -d)
+TEMP_CONFIG_FILE="$TEMP_DIR/config.mk"
 
-# copy the common (all platforms) PcapPlusPlus.mk
-cp -f mk/PcapPlusPlus.mk.common $PCAPPLUSPLUS_MK
+cat > "$TEMP_CONFIG_FILE" <<EOF
+PCAPPLUSPLUS_HOME   := $SCRIPT_DIR
+ANDROID_TARGET      := $TARGET_TRIPLE$ANDROID_API_LEVEL
+ANDROID_NDK_PATH    := $ANDROID_NDK_ROOT
```

Comment:
I don't think we need this variable anymore, it's not used anywhere. Should we remove it?

Reply:
I find the guide. 
https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-base-branch-of-a-pull-request

Reply:
ok, last step, remove the used variable.